### PR TITLE
Adding section on unknown discriminants

### PR DESCRIPTION
**Pull request by gusthoff**

### CODE REVIEW

File: content/courses/advanced-ada/parts/data_types/records.rst [hunk 1]
```diff
@@ -2124,19 +2124,506 @@ value set for the parent type :ada:`T`.
...
+
+       type Derived_Rec is
+         new Rec;
+
+    end Unknown_Discriminants.Children;
+
+Note that :ada:`Derived_Rec` has unknown discriminants, even though we're not
+explicitly using an unknown discriminant part (:ada:`(<>)`) in its declaration.
+(In fact, we're not allowed to use an unknown discriminant part in this case.)
+Therefore, declaring objects of this type directly isn't possible:
```
Comment (pat-rogers) [inline content/courses/advanced-ada/parts/data_types/records.rst:2402]: I think you might want to emphasize that declaring objects of type Derived_Rec remains impossible, like the parent type Rec. My concern is that the reader might think that this is a new restriction, introduced by the derivation.
So maybe change
"directly isn't possible:"
to
"directly isn't possible, just like the parent type Rec:"

File: content/courses/advanced-ada/parts/data_types/records.rst [hunk 2]
```diff
@@ -2124,19 +2124,506 @@ value set for the parent type :ada:`T`.
...
+       type Derived_Rec_No_Discr is
+         new Rec with null record;
+
+    end Unknown_Discriminants.Children;
+
+In this example, we declare :ada:`Derived_Rec_Unknown_Discr` with an unknown
+discriminant part, :ada:`Derived_Rec_Known_Discr` with a known discriminant
+part, and :ada:`Derived_Rec_No_Discr` with no discriminants.
+
+As expect, :ada:`Derived_Rec_Unknown_Discr` has unknown discriminants because
```
Comment (pat-rogers) [inline content/courses/advanced-ada/parts/data_types/records.rst:2467]: "As expect" => "As expected"

File: content/courses/advanced-ada/parts/resource_management/limited_types.rst [hunk 3]
```diff
@@ -1772,15 +1772,10 @@ the :ada:`Init` function.)
 Limited private type with unknown discriminants
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-We can declare limited private type with unknown discriminants. Because the
-discriminants are unknown, this is an
-:ref:`indefinite type <Adv_Ada_Definite_Indefinite_Subtypes>`. Let's see an
+We can declare limited private type with
```
Comment (pat-rogers) [inline content/courses/advanced-ada/parts/resource_management/limited_types.rst:1775]: "declare limited private type with"  =>  "declare a limited private type with"
or
"declare limited private types with"

File: content/courses/advanced-ada/parts/data_types/records.rst (hunk 1 above)
Comment (gusthoff) [reply content/courses/advanced-ada/parts/data_types/records.rst]: Yes, that makes sense. Thanks for the suggestion!

**Review by pat-rogers — APPROVED** [review]

**Pull request merged**